File: MoveZeroes/MZ.java
class Solution {
    public void moveZeroes(int[] nums) {
        int i =0;
        int j =1;
        while(j<nums.length){
            if(nums[i] == 0 && nums[j]!=0){
                int temp = nums[j];
                nums[j] = nums[i];
                nums[i] = temp;
                i++;
                j++;
            }
            else if(nums[i]==0 && nums[j]== 0){
                j++;
            }
            else if(nums[i]!=0){
                i++;
                j++;
            }
        }
        
        
    }
}

File: MoveZeroes/MZ.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        i,j = 0,1
        while j<len(nums):
            if nums[i]==0 and nums[j]!=0:
                temp = nums[j]
                nums[j] = nums[i]
                nums[i] = temp
                i+=1
                j+=1
            elif nums[i]==0 and nums[j]==0:
                j+=1
            elif nums[i]!=0:
                i+=1
                j+=1
        